content/client-lib-development-guide/features.textile

```diff
@@ -1473,6 +1495,82 @@ enum StatsIntervalGranularity:
   DAY
   MONTH
 
+class DeviceDetails:
+  + fromLocalDevice() => io DeviceDetails
+  id: String
+  platform: DevicePlatform
+  formFactor: DeviceFormFactor
+  clientId: String?
+  metadata: Dict<String, String>
+  updateTokenId: String
+  resetId(String)
```

Comment:
The original reason was that you might want to inherit all the server-side state if e. g. you reinstalled the app, and not do it by client ID. But yes, I think we agreed on not allowing this in the end. Fixed at 1d7571245a90a5d9fe927d0f4c68ab8eb03081b3.